engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -482,23 +546,20 @@ class JoinPartitionKernel :public kernel {
 			}
 		}
 
-		BlazingMutableThread distribute_left_thread(&JoinPartitionKernel::partition_table, context, 
+		BlazingMutableThread distribute_left_thread(&JoinPartitionKernel::partition_table, this->context, 
 			this->left_column_indices, std::move(left_batch), std::ref(left_sequence), 
 			std::ref(this->output_.get_cache("output_a")));
 
 		BlazingThread left_consumer([context = this->context, this](){
 			auto  message_token = ColumnDataPartitionMessage::MessageID() + "_" + this->context->getContextCommunicationToken();
 			ExternalBatchColumnDataSequence external_input_left(this->context, message_token);
-			std::unique_ptr<ral::frame::BlazingHostTable> host_table;
-			//std::cout << "... consumming left => "<< this->get_message_id()<<std::endl;
-			while (host_table = external_input_left.next()) {	
+			
+			while (external_input_left.wait_for_next()) {	
```

Comment:
Before this was a synchronous operation because you had one thread getting the next oen without checking if there is a next, What is the logic behind splitting this up into two functions when what you want to know is:

Do you have one ready for me if so give it to me if not then the ideal would be to wait to check until it should check again using something like a condition variable. Either way splitting it up into two pieces means if there are multiple threads consuming at some point for some reason that you could get a next is available signal and then not be able to actually get it when the time comes.

Reply:
Hmmm, i had not thought of that. I see what you mean. In this case, there is only one thread accessing that external_input_left, but i will change it back to make that the more standard way of doing it.